File: 1983.py
alunos = int(input())
melhor_nota = 0
matricula_melhor_nota = ""

for i in range(alunos):
    entrada = input().split()
    matricula = entrada[0]
    nota = float(entrada[1])
    
    if (i == 0 or nota > melhor_nota):
        melhor_nota = nota
        matricula_melhor_nota = matricula

if (melhor_nota >= 8):
    print(matricula_melhor_nota)
else:
    print("Minimum note not reached")
   

File: 1983.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

var alunos = Number(lines.shift().trim());
var melhor_nota;
var matricula_melhor_nota;
var i = 0;

while (i < alunos) {
    var entrada = lines[i].split(" ");
    var matricula = entrada[0];
    var nota = parseFloat(entrada[1]);
    
    if (i === 0 || nota > melhor_nota) {
        melhor_nota = nota;
        matricula_melhor_nota = matricula;
    }
    
    i++;
}

if (melhor_nota < 8) {
    console.log("Minimum note not reached");
} else {
    console.log(matricula_melhor_nota);
}
